swagger-doclet/src/main/java/com/tenxerconsulting/swagger/doclet/parser/ParserHelper.java

```diff
@@ -163,6 +173,35 @@ public static String getInheritableClassLevelAnnotationValue(ClassDoc classDoc,
         }
         return null;
     }
+    
+    /**
+     * This gets an annotation value array from the given class, it supports looking at super classes
+     * @param classDoc The class to look for the annotation
+     * @param options The doclet options
+     * @param qualifiedAnnotationType The FQN of the annotation to look for
+     * @param keys The keys for the annotation values
+     * @return The value of the annotation or null if none was found
+     */
+    public static String[] getInheritableClassLevelAnnotationValues(ClassDoc classDoc, DocletOptions options, String qualifiedAnnotationType, String... keys) {
+        ClassDoc currentClassDoc = classDoc;
+        while (currentClassDoc != null) {
+
+            AnnotationParser p = new AnnotationParser(currentClassDoc, options);
+            String[] values = p.getAnnotationValues(qualifiedAnnotationType, keys);
+
+            if (values != null) {
+                return values;
+            }
+
+            currentClassDoc = currentClassDoc.superclass();
+            // ignore parent object class
```

Comment:
Why do we want to ignore a root parent type? Wouldn't that include an implemented interface?

Reply:
Donno. Copied it from the block above. It's seems to be a null check and just skips java.lang.Object. Not pretty, but ok, I think.